ddl/ddl_worker.go

```diff
@@ -248,6 +248,18 @@ func (d *ddl) runDDLJob(t *meta.Meta, job *model.Job) (ver int64) {
 	if job.IsFinished() {
 		return
 	}
+	// The cause of this job state is that the job is cancelled by client.
+	if job.IsCancelling() {
+		// If the value of SnapshotVer isn't zero, it means the work is backfilling the indexes.
+		if job.Type == model.ActionAddIndex && job.SchemaState == model.StateWriteReorganization && job.SnapshotVer != 0 {
+			d.notifyCancelReorgJob <- struct{}{}
```

Comment:
How about ActionDropIndex?
And I'm not sure if there is a problem but use a `select` to send notify channel is safer.

Reply:
how about use a map or array like:

```
m := map[int]bool{
	model.ActionAddIndex: true,
	...
}
```
to manager which job type will do reorganize. and avoid omission.

Reply:
Now we only one action needs to do reorganize, so we don't need a map or a array.